### DISCUSSION TITLE
Unable to Run "prisma generate" during post-install script

### DISCUSSION THREAD

**Discussion by wtseng21** [MichalLytek/typegraphql-prisma#269] [Q&A]
Hello. I'm having trouble trying to get "prisma generate" to run during the post-install script after running `npm ci`. 

My **schema.prisma** contains the following:
```
generator client {
  provider = "prisma-client-js"
}

generator typegraphql {
  provider = "typegraphql-prisma"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

I'm getting the message:

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
info The postinstall script automatically ran `prisma generate`, which failed.
The postinstall script still succeeds but won't generate the Prisma Client.
Please run `prisma generate` to see the errors.
```

Running `npx prisma generate` manually does not have any issues.
If I remove the `generator typegraphql` code snippet, then the message does not pop up.

If it helps, I tried installing `prisma` globally as well, and when I attempt to do `prisma generate`, I get the following error:
```
Error: Generator at typegraphql-prisma could not start:

/bin/sh: typegraphql-prisma: command not found
```

Not sure if I missed a step in the guide? Thanks!

**Comment by MichalLytek** [comment]
Have you installed the npm package?

**Reply by wtseng21** [reply]
Yeah, here's the `package.json`

```
  "dependencies": {
    ...
    "@prisma/client": "^3.11.1",
    "@types/graphql-fields": "^1.3.4",
    "@types/validator": "^13.7.2",
    "apollo-server": "^3.6.6",
    "class-validator": "^0.13.2",
    "dotenv": "^2.0.0",
    "graphql": "^15.3.0",
    "graphql-fields": "^2.0.3",
    "graphql-scalars": "^1.17.0",
    "reflect-metadata": "^0.1.13",
    "tslib": "^2.3.1",
    "type-graphql": "^1.1.1"
  },
  "devDependencies": {
    ...
    "prisma": "^3.11.1",
    "ts-node": "^10.7.0",
    "ts-node-dev": "^1.1.8",
    "typegraphql-prisma": "^0.19.2",
    "typescript": "^4.6.3"
  },
```

This kind of behavior seems consistent with other generators for me. Is it generally not expected that the `@prisma/client` will be reinstalled during, for example, deployment? Or is it expected to just run another dedicated step to generate.

**Reply by MichalLytek** [reply]
If it's consistent, then I don't know if I can do anything about this 😛 
For my projects it works without any issues on post install.

**Reply by wtseng21** [reply]
Wondering if it’s an OS issue…
Do you have the prisma and generator packages installed globally or locally? 

EDIT: Upon further inspection, it seems like this issue only occurs when doing `npm ci`. When manually removing **node_modules** and **package-lock.json** and running `npm i` afterwards, it works fine.

**Reply by wtseng21** [reply]
Following up, it looks like installing `typegraphql-prisma` as a `dependency` rather than a `devDependency` seems to resolve the issue.
